### PR TITLE
add transport_opts to data migration

### PR DESCRIPTION
**Pull request by ruslandoga**
### Changes

Closes https://github.com/plausible/analytics/issues/3173

### Tests
- [x] This PR does not require tests

### Changelog
- [x] Entry has been added to changelog

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: lib/plausible/data_migration/repo.ex [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
-  Ecto.Repo for Clickhouse data migrations, to be started manually, 
+  Ecto.Repo for Clickhouse data migrations, to be started manually,
   outside of the main application supervision tree.
   """
   use Ecto.Repo,
     otp_app: :plausible,
     adapter: Ecto.Adapters.ClickHouse
 
   def start(url, max_threads) when is_binary(url) and is_integer(max_threads) do
+    default_config = Plausible.IngestRepo.config()
```
Comment (ruslandoga) [inline lib/plausible/data_migration/repo.ex:11]: I'll need to try it out locally now to make sure `Plausible.IngestRepo.config()` is available when running the migration.

I'm using IngestRepo to be consistent with https://github.com/plausible/analytics/blob/16846b16c8e513f72d7c23459ee879d9249f556d/lib/plausible/data_migration/numeric_ids.ex#L30

**Review by aerosol — APPROVED** [review]

**Comment by aerosol** [comment]
Hmm this is weird: https://github.com/plausible/analytics/actions/runs/5641146895/job/15278756140?pr=3179

**Comment by aerosol** [comment]
> Hmm this is weird: https://github.com/plausible/analytics/actions/runs/5641146895/job/15278756140?pr=3179

is it because you're submitting the PR off your fork? 🤔

**Comment by ruslandoga** [comment]
Yeah, I think so.

**Pull request merged**